src/components/HashButton.tsx

```diff
@@ -13,12 +13,51 @@ import {grey} from "../themes/colors/aptosColorPalette";
 import ChevronRightRoundedIcon from "@mui/icons-material/ChevronRightRounded";
 import ChevronLeftRoundedIcon from "@mui/icons-material/ChevronLeftRounded";
 import {truncateAddress} from "../pages/utils";
+import {assertNever} from "../utils";
+
+export enum HashType {
+  ACCOUNT = "account",
+  TRANSACTION = "transaction",
+  OTHERS = "others",
+}
+
+function HashLink(hash: string, type: HashType): JSX.Element {
+  switch (type) {
```

Comment:
why not just passing the url? (i.e `account/txn`) that way dont need to create new element (that are pretty much the same) for each "type"? 

```suggestion
 function HashLink(hash,url){
 return (
 <Link
          component={RRD.Link}
          to={`/${url}/${hash}`}
          target="_blank"
          color="inherit"
        >
          {hash}
        </Link>
 )
 }
```

Reply:
or if you prefer keeping a type, can hold a `type=>url` map and apply the url to the Link component based on the type